build.assets/tooling/cmd/release-notes/release_notes.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"html/template"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+var (
+	versionMatcher      = regexp.MustCompile(`\d+\.\d+\.\d+`)
+	doubleHeaderMatcher = regexp.MustCompile(`^##\s+`)
+)
+
+//go:embed template/release-notes.md.tmpl
+var tmpl string
+
+type tmplInfo struct {
+	Version     string
+	Description string
+}
+
+var (
+	releaseNotesTemplate = template.Must(template.New("release notes").Parse(tmpl))
+)
+
+type releaseNotesGenerator struct {
+	// releaseVersion is the version for the release.
+	// This will be compared against the version present in the changelog.
+	releaseVersion string
+}
+
+func (r *releaseNotesGenerator) generateReleaseNotes(md *os.File) (string, error) {
+	desc, err := r.parseMD(md)
+	if err != nil {
+		return "", err
+	}
+
+	info := tmplInfo{
+		Version:     r.releaseVersion,
+		Description: desc,
+	}
+	var buff bytes.Buffer
+	if err := releaseNotesTemplate.Execute(&buff, info); err != nil {
+		return "", trace.Wrap(err)
+	}
+	return buff.String(), nil
+}
+
+// parseMD is a simple implementation of a parser to extract the description from a changelog.
+// Will scan for the first double header and pull the version from that.
+// Will pull all information between the first and second double header for the description.
+func (r *releaseNotesGenerator) parseMD(md *os.File) (string, error) {
```

Comment:
This should take an `io.Reader` as a parameter, not `*os.File` as `io.Reader` is all that it needs. It then allows testing with other simulated input not from files. Same for `generateReleaseNotes()` above.